node/src/reactor/validator.rs

```diff
@@ -721,8 +721,12 @@ impl reactor::Reactor for Reactor {
             }) => {
                 let event = block_proposer::Event::BufferDeploy {
                     hash: *deploy.id(),
-                    header: Box::new(deploy.header().clone()),
-                    is_transfer: matches!(deploy.session(), ExecutableDeployItem::Transfer{ .. }),
+                    deploy_type: match deploy.session() {
+                        ExecutableDeployItem::Transfer { .. } => Box::new(
+                            block_proposer::DeployType::Transfer(deploy.header().clone()),
+                        ),
+                        _ => Box::new(block_proposer::DeployType::Wasm(deploy.header().clone())),
```

Comment:
I'd avoid `_` in general where possible, just so we don't forget anything if we add variants.

But can we get rid of these lines altogether, and add a method so that we can just use something like
```rust
    deploy_type: deploy.deploy_type(),
```
here?

Reply:
Well, `deploy` here is a `DeployHeader`, so we'd have to pass the type that it should become. I'm not sure that reduces this overhead.